Protocol handlers without web+ prefix in Firefox 59

lidel opened the pull request:
This PR implements changes described in https://github.com/ipfs-shipyard/ipfs-companion/issues/164#issuecomment-355708883, namely support for simplified (redirect-based) protocol handlers for  `dweb:` and `ipfs://` without `web+` prefix.  

It works perfectly fine with latest Firefox Nightly (59a1), but fails to load in Firefox <59:

```
$ web-ext run -s add-on/ --browser-console --verbose
[firefox/index.js][debug] Firefox stdout: 1516628484552	addons.webextension.<unknown>	ERROR	Loading extension 'null': Reading manifest: Error processing protocol_handlers.3.protocol: Value "dweb" must either: be one of ["bitcoin", "geo", "gopher", "im", "irc", "ircs", "magnet", "mailto", "mms", "news", "nntp", "sip", "sms", "smsto", "ssh", "tel", "urn", "webcal", "wtai", "xmpp"], or match the pattern /^(ext|web)\+[a-z0-9.+-]+$/
```

Due to this and a false-positive error in addon-linter (https://github.com/mozilla/addons-linter/issues/1783) I am parking this PR until Firefox 59 is released to Beta (it may even wait til Stable).

lidel commented:
I rebased this PR to use the latest `web-ext` that includes `addon-linter`  with updated rules (https://github.com/mozilla/addons-linter/issues/1783).  This means `npm run lint` no longer breaks our tests 👌 

Unfortunately if we enable non-prefixed handlers in `manifest.json` Firefox < 59 will refuse to load extension at all (Chrome is just ignoring unknown keys and runs fine). 

That is why I am parking this PR until Firefox 59 becomes the new Stable.

chpio commented:
> Version 59.0, first offered to Release channel users on March 13, 2018
> \- https://www.mozilla.org/en-US/firefox/59.0/releasenotes/

lidel commented:
Will rebase and merge this ASAP, so that we have a new FF59-enabled beta release :)

lidel commented:
Rebased and smoke-tested on Linux with Firefox 59, works as expected. 


### Build for Firefox 58

Non-prefixed protocol handlers in manifest break Firefox 58 (extension just does not load) so to ease transition for casual contributors and testers I've added an optional target that builds bundle for the old version without this new feature:
```
npm run build bundle:firefox:58
```

Keep in mind this target is temporary, disabled by default and will be removed in 2 weeks.
Upgrade to Firefox 59 ASAP.

lidel commented:
If anyone has trouble with downloading Firefox 59 there seems to be an upstream problem:  

- https://bugzilla.mozilla.org/show_bug.cgi?id=1445672
